cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -176,36 +185,52 @@ static Bom createBom() {
-                        // read the sbom file as a json object writer
-                        JSONObjectWriter sbom = new JSONObjectWriter(new FileInputStream(sbomFile));
-                        // sign the sbom
-                        byte[] signedSbom = signer.signData(sbom.getEncoded());
-                        // Write the signed SBOM to a new file
-                        FileOutputStream signedSbomStream = new FileOutputStream("signed_" + sbomFile.getName());
-                        signedSbomStream.write(signedSbom);
-                        signedSbomStream.close();
-                        System.out.println("SBOM " + sbomFile.getName() + " signed and saved as signed_" + sbomFile.getName());
-                    }
-                }
-            } else {
-                System.out.println("Error: the home directory is empty or does not exist.");
-                System.exit(1);
-            }
-            // Print the certificate information
-            System.out.println("SBOMs signed with certificate: " + signer.getCertificate().toString());
-        } catch (Exception e) {
-            System.out.println("Error signing SBOM: " + e.getMessage());
+            // Read multiple JSON files to be signed
+            File[] jsonFiles = new File[n]; // where n is the number of files to be read
+            byte[][] jsonData = new byte[n][];
+            String[] sbomDataToSign = new String[n];
+            for (int i = 0; i < n; i++) {
+            FileInputStream jsonFileInputStream = new FileInputStream(jsonFiles[i]);
+            jsonData[i] = new byte[(int) jsonFiles[i].length()];
+            jsonFileInputStream.read(jsonData[i]);
+            jsonFileInputStream.close();
+            sbomDataToSign[i] = new String(jsonData[i], "UTF-8");
+        }
+
+            // Read the private key
+            File privateKeyFile = new File("RSAPrivate.pem");
+            byte[] privateKeyData = new byte[(int) privateKeyFile.length()];
+            FileInputStream privateKeyFileInputStream = new FileInputStream(privateKeyFile);
+            privateKeyFileInputStream.read(privateKeyData);
+            privateKeyFileInputStream.close();
+            String privateKeyString = new String(privateKeyData, "UTF-8");
+            KeyPair sampleKey = PEMDecoder.getKeyPair(privateKeyData);
+
+
+            // Sign the JSON data
+            String signedData = new JSONObjectWriter(JSONParser.parse(sbomDataToSign))
+                    .setSignature(new JSONAsymKeySigner(sampleKey.getPrivate()))
+                    .serializeToString(JSONOutputFormats.PRETTY_PRINT);
+
+            // Append the signature to the original JSON data
+            sbomDataToSign = sbomDataToSign.substring(0, sbomDataToSign.lastIndexOf('}')) +
+                    "," +
+                    signedData.substring(signedData.indexOf("\n  \"signature"));
+
+            // Read the public key
+            File publicKeyFile = new File("RSAPrivate.pem");
```

Comment:
The pem file just contains a private key, you're effectively printing out the private key, which would be a security exposure !
If we want to print out a public key you would need to use openssl to generate a public key from the private one.
However, we don't need to print this here, simply remove these lines